### ISSUE TITLE
process howto uses indentation in pre tags incorrectly

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3632]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3632|
|Type|defect|
|Created|2009-01-24 00:13:49Z|
|Branch|https://github.com/twisted/twisted/tree/process-docs-pre-usage-3632|

It's basically always incorrect to have base indention inside a `<pre>` tag, and this is true in the case of a Lore input document's `<pre class="python>` as well.

The least indented code in a Python snippet must have no indentation at all, otherwise there are unpleasant rendering consequences.

The resulting problem can be observed by processing the document using lore then looking at one of the code listings which uses indentation incorrectly (for example, the stand-alone definition of `connectionMade`).  Viewed in an HTML renderer, the second and subsequent lines will be indented too much (or conversely, the first line will not be indented enough).

<details><summary>Searchable metadata</summary>

```
trac-id__3632 3632
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_process_docs_pre_usage_3632 branches/process-docs-pre-usage-3632
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation_easy documentation easy
time__1232756029000000 1232756029000000
changetime__1234666260000000 1234666260000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [26196]) Branching to 'process-docs-pre-usage-3632'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [26197]) dedent

refs [#3632](https://github.com/twisted/twisted/issues/3632)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

[Build Results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/process-docs-pre-usage-3632&builder=documentation)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Hum ok, +1.

I'm tempted to think that it's a bug in lore though (the first line should be more indented). It would help copy/paste, for example. That can be fixed later, though.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [26223]) Merge process-docs-pre-usage-3632

Author: exarkun
Reviewer: therve
Fixes: [#3632](https://github.com/twisted/twisted/issues/3632)

Dedent the Python code samples in the process howto.  The extra indentation causes the source
to be misrendered (with not enough indentation on the first line when displayed).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
